### PR TITLE
Revert back to e2e sh

### PR DESCRIPTION
**Pull request by ayj**
* Revert Remove e2e.sh and change default tag to <username>_<timestamp> (#140)
* Update bin/e2e.sh to default tag of <username>_<timestamp>
* Update bin/e2e.sh to pass unrecognized test arguments through to driver.go

### CODE REVIEW

File: test/integration/driver.go [hunk 1]
```diff
@@ -83,16 +82,14 @@ func init() {
 		"kube config file or empty for in-cluster")
 	flag.StringVarP(&hub, "hub", "h", "gcr.io/istio-testing",
 		"Docker hub")
-	flag.StringVarP(&tag, "tag", "t", "",
-		"Docker tag (default <username>_YYYMMDD_HHMMSS formatted date)")
+	flag.StringVarP(&tag, "tag", "t", "test",
+		"Docker tag")
```
Comment (rshriram) [inline test/integration/driver.go]: Don't revert this. This doesn't match with the shell script. And we have abused test tags a lot already ;)

File: bin/e2e.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+set -ex
+
+if [[ "$hub" =~ ^gcr\.io ]]; then
+    gcloud docker --authorize-only
+fi
+
+for image in app init runtime; do
+    bazel run //docker:$image
+    docker tag istio/docker:$image $hub/$image:$tag
```
Comment (rshriram) [inline bin/e2e.sh]: What happened to that docker save trick? I thought it helped

File: test/integration/driver.go (hunk 1 above)
Comment (ayj) [inline test/integration/driver.go]: driver.go no longers builds/tags/pushes the docker images. I moved this all back to bin/e2e.sh which can invoke 'bazel run' directly and avoid the issues we were seeing with missing layers/images in Jenkins.

File: bin/e2e.sh (hunk 2 above)
Comment (ayj) [inline bin/e2e.sh]: No, the save+import thing never worked. Loading the intermediate layers (e.g. bazel-bin/docker/app-layer.tar) worked locally but failed in Jenkins.

File: test/integration/driver.go (hunk 1 above)
Comment (ayj) [inline test/integration/driver.go]: I removed the default value and made the user specify tag explicitly to avoid mismatch with bin/e2e.sh. PTAL

**Pull request merged**